File: patterns/c-patterns/pattern21.c
// Pattern 21: ...

// A****
// AB***
// ABC**
// ABCD*
// ABCDE

#include <stdio.h>
int main()
{
    int i, j;
    int n = 5;
    for (i = 0; i < n; i++)
    {
        for (j = 0; j < n; j++)
        {
            if (j > i)
                printf("*");
            else
                printf("%c", j + 65);
        }
        printf("\n");
    }
    return 0;
}

File: patterns/python-patterns/pattern21.py
# Pattern 21: ...

# A****
# AB***
# ABC**
# ABCD*
# ABCDE

n = 5

for i in range(n):
    for j in range(n):
        if j > i:
            print("*", end="")
        else:
            print(chr(j + 65), end="")
    print()
